File: Prácticas1DAW/Ficheros/ficheros/Ejercicio1.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
Escribe un programa que guarde en un escritura con
nombre primos.dat los números primos que hay entre 1 y 500.
"""

#Creamos un archivo llamado primos.dat y lo abrimos en modo escritura
escritura = open("primos.dat","w")


def esPrimo(num):
    """
    Esta función comprueba si un número es primo
    :param num:
    :return: devuelve True si el número pasado por prámetro es primo
    """
    if num <= 1:
        return False
    elif num == 2:
        return True
    else:
        for i in range(2, num):
            if num % i == 0:
                return False
        return True

for x in range(501):
    """
    Con este for compruebo los números primos del 1 al 500 y los añado al escritura antes creado
    """
    if(esPrimo(x) == True):
        escritura.writelines(str(x)+" ")

#Cierro el flujo de datos 
escritura.close()

File: Prácticas1DAW/Ficheros/ficherosJava/src/ficheros/Ejercicio1.java
package ficheros;
/**
 * Escribe un programa que guarde en un fichero con
 * nombre primos.dat los números primos que hay entre 1 y 500.
 * @author juanbu97
 */

import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;

public class Ejercicio1 {

  public static boolean esPrimo(int x) {

    boolean esPrimo = true;

    int contador = 2;
    while ((esPrimo) && (contador != x)) {
      if (x % contador == 0)
        esPrimo = false;
      contador++;
    }

    return esPrimo;

  }

  public static void main(String[] args) {
    // System.out.println(esPrimo(7));

    try {
      BufferedWriter escribe = new BufferedWriter(new FileWriter("primos.dat"));

      for (int i = 0; i < 501; i++) {
        if (esPrimo(i) == true)
          escribe.write(" "+i+" \n");
      }
      
      escribe.close();

    } catch (IOException e) {
      System.err.println("No se ha podido escribir en el fichero");
    }

  }
}
